catalog/ui/search-ui/standard/src/main/java/org/codice/ddf/ui/searchui/standard/properties/ConfigurationStore.java

```diff
@@ -265,22 +259,75 @@ public void setTerrainProvider(String terrainProvider) {
         setProxyForTerrain(terrainProvider);
     }
 
-    private void setProxiesForImagery(List<String> imageryProviders) {
-        if (imageryEndpoints.size() > 0) {
-            for (String endpoint : imageryEndpoints) {
-                try {
-                    httpProxy.stop(endpoint);
-                } catch (Exception e) {
-                    LOGGER.error("Unable to stop proxy endpoint.", e);
+    private void setProxiesForImagery(List<String> proxiesForImagery) {
+        List<Map<String, Object>> newImageryProviders = proxiesForImagery.stream()
+                .map(this::getProviderConfigMap)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+        List<Map<String, Object>> imageryProvidersToStop = new ArrayList<>();
+        List<Map<String, Object>> imageryProvidersToStart = new ArrayList<>();
+
+        findDifferences(imageryProviderMaps, newImageryProviders, imageryProvidersToStart);
+        findDifferences(newImageryProviders, imageryProviderMaps, imageryProvidersToStop);
+
+        List<String> proxiesToStop = imageryProvidersToStop.stream()
```

Comment:
I'm confident that it is always a String. The Object thing is just an artifact of that library we're using to do EDN.